aci/resource_aci_fvesg.go

```diff
@@ -0,0 +1,696 @@
+		err = aciClient.CreateRelationfvRsScopeFromEndpointSecurityGroup(fvESg.DistinguishedName, relationParamName)
+		if err != nil {
+			return err
+		}
+		d.Partial(true)
+		d.SetPartial("relation_fv_rs_scope")
+		d.Partial(false)
+
+	}
+	if relationTofvRsProtBy, ok := d.GetOk("relation_fv_rs_prot_by"); ok {
+		relationParamList := toStringList(relationTofvRsProtBy.(*schema.Set).List())
+		for _, relationParam := range relationParamList {
+			relationParamName := GetMOName(relationParam)
+			err = aciClient.CreateRelationfvRsProtByFromEndpointSecurityGroup(fvESg.DistinguishedName, relationParamName)
+
+			if err != nil {
+				return err
+			}
+			d.Partial(true)
+			d.SetPartial("relation_fv_rs_prot_by")
+			d.Partial(false)
+		}
+	}
+	if relationTofvRsIntraEpg, ok := d.GetOk("relation_fv_rs_intra_epg"); ok {
+		relationParamList := toStringList(relationTofvRsIntraEpg.(*schema.Set).List())
+		for _, relationParam := range relationParamList {
+			relationParamName := GetMOName(relationParam)
+			err = aciClient.CreateRelationfvRsIntraEpgFromEndpointSecurityGroup(fvESg.DistinguishedName, relationParamName)
+
+			if err != nil {
+				return err
+			}
+			d.Partial(true)
+			d.SetPartial("relation_fv_rs_intra_epg")
+			d.Partial(false)
+		}
+	}
+
+	d.SetId(fvESg.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+
+	return resourceAciEndpointSecurityGroupRead(d, m)
+}
+
+func resourceAciEndpointSecurityGroupUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] EndpointSecurityGroup: Beginning Update")
+
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+
+	ApplicationProfileDn := d.Get("application_profile_dn").(string)
+
+	fvESgAttr := models.EndpointSecurityGroupAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		fvESgAttr.Annotation = Annotation.(string)
+	}
+	if ExceptionTag, ok := d.GetOk("exception_tag"); ok {
+		fvESgAttr.ExceptionTag = ExceptionTag.(string)
+	}
+	if FloodOnEncap, ok := d.GetOk("flood_on_encap"); ok {
+		fvESgAttr.FloodOnEncap = FloodOnEncap.(string)
+	}
+	if MatchT, ok := d.GetOk("match_t"); ok {
+		fvESgAttr.MatchT = MatchT.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		fvESgAttr.NameAlias = NameAlias.(string)
+	}
+	if PcEnfPref, ok := d.GetOk("pc_enf_pref"); ok {
+		fvESgAttr.PcEnfPref = PcEnfPref.(string)
+	}
+	if PrefGrMemb, ok := d.GetOk("pref_gr_memb"); ok {
+		fvESgAttr.PrefGrMemb = PrefGrMemb.(string)
+	}
+	if Prio, ok := d.GetOk("prio"); ok {
+		fvESgAttr.Prio = Prio.(string)
+	}
+	if Userdom, ok := d.GetOk("userdom"); ok {
+		fvESgAttr.Userdom = Userdom.(string)
+	}
+	fvESg := models.NewEndpointSecurityGroup(fmt.Sprintf("esg-%s", name), ApplicationProfileDn, desc, fvESgAttr)
+
+	fvESg.Status = "modified"
+
+	err := aciClient.Save(fvESg)
+
+	if err != nil {
+		return err
+	}
+	d.Partial(true)
+
+	d.SetPartial("name")
+
+	d.Partial(false)
```

Comment:
There should one pre check for all relations. That will check whether relation DN actually exist or not.

Reply:
Can you please elaborate on the pre-check condition and give an example?
Thanks.